mapper_solution.ml

```diff
@@ -16,6 +16,21 @@ let rec remove_type_annotations = function
         Location.error ~loc "Not a function or lacking type annotations."))
 
 let solution_structure_mapper mapper s =
+  let filter = function
```

Comment:
This choice of identifiers is a bit... let's say... confusing!

Reply:
Changed it. Is the version with map and filter better? My first idea was to combine both to avoid going through the list and pattern-match twice.

Reply:
This is also what I do on many other similar functions.

Reply:
FP motto: Writing small and clear-as-cristal functions is usually better than writing a single monolithic complex one. 

Another motto: Premature optimization is the root of all evil.